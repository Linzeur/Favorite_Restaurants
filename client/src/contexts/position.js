import React from "react";

const PositionContext = React.createContext();

function PositionProvider(props) {
  const [position, setPosition] = React.useState([0, 0, false]);

  React.useEffect(() => {
    const watchId = navigator.geolocation.watchPosition(({ coords }) => {
      setPosition([coords.latitude, coords.longitude, true]);
      sessionStorage.setItem(
        "lastCurrentPosition",
        JSON.stringify({ lat: coords.latitude, lng: coords.longitude })
      );
    });
    return () => navigator.geolocation.clearWatch(watchId);
  }, [setPosition]);

  const value = {
    latitude: position[0],
    longitude: position[1],
    callAPI: position[2],
    onChange: (lat, lng, callAPI) => {
      setPosition([lat, lng, callAPI]);
    }
  };

  return <PositionContext.Provider value={value} {...props} />;
}

export { PositionProvider, PositionContext };
