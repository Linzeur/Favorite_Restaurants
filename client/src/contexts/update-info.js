import React from "react";

const UpdateListContext = React.createContext();

function UpdateListProvider(props) {
  const [updateList, setUpdateList] = React.useState({
    locations: 0,
    favorites: 0
  });

  const value = {
    locations: updateList.locations,
    favorites: updateList.favorites,
    onChange: key => {
      setUpdateList(prevState => {
        return { ...prevState, [key]: prevState[key] + 1 };
      });
    }
  };

  return <UpdateListContext.Provider value={value} {...props} />;
}

export { UpdateListProvider, UpdateListContext };
