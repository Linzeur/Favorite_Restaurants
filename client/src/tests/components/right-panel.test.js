import React from "react";
import { render, fireEvent } from "@testing-library/react";

import RightPanel from "../../components/right-panel";
import { PositionProvider } from "../../contexts/position";
import { UpdateListProvider } from "../../contexts/update-info";

test("right-panel component", () => {
  const { asFragment } = render(
    <PositionProvider>
      <UpdateListProvider>
        <RightPanel />
      </UpdateListProvider>
    </PositionProvider>
  );

  expect(asFragment()).toMatchSnapshot();
});

test("click in Current Location button", () => {
  const { getAllByRole } = render(
    <PositionProvider>
      <UpdateListProvider>
        <RightPanel />
      </UpdateListProvider>
    </PositionProvider>
  );

  const buttonCurrentLocation = getAllByRole("button")[0];
  fireEvent.click(buttonCurrentLocation);
  const coords = JSON.parse(sessionStorage.getItem("lastCurrentPosition"));

  expect(coords.lat).toBe(-11.9913905);
  expect(coords.lng).toBe(-77.0873197);
});
