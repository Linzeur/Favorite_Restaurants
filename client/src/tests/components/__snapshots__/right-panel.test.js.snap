// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`right-panel component 1`] = `
<DocumentFragment>
  <div
    class="css-q05jr7-RightPanel"
  >
    <button
      class="css-1cabchf-Button"
      style="padding: 0.5em;"
      type="button"
    >
      Current Location
    </button>
    <div
      data-reach-tabs=""
    >
      <div
        class="css-mowv2j-RightPanel"
        data-reach-tab-list=""
        role="tablist"
      >
        <button
          aria-controls="panel:1:0"
          aria-selected="true"
          class="css-1n5xkbe-RightPanel"
          data-reach-tab=""
          data-selected=""
          id="tab:1:0"
          role="tab"
          tabindex="0"
        >
          My Locations
        </button>
        <button
          aria-controls="panel:1:1"
          aria-selected="false"
          class="css-1n5xkbe-RightPanel"
          data-reach-tab=""
          id="tab:1:1"
          role="tab"
          tabindex="-1"
        >
          My Favorites
        </button>
      </div>
      <div
        class="css-1upy4sj-RightPanel"
        data-reach-tab-panels=""
      >
        <div
          aria-labelledby="tab:1:0"
          class="css-818u9c-RightPanel"
          data-reach-tab-panel=""
          id="panel:1:0"
          role="tabpanel"
          tabindex="-1"
        >
          <div
            class="css-1ylqutl-Search"
          >
            <input
              aria-label="Search an option"
              class="css-x0timv-Search"
              id="search"
              name="search"
              placeholder="Search an option"
              type="text"
            />
            <button
              aria-label="Create location"
              class="css-1mgkve5-AddLocation"
              type="button"
            >
              <svg
                class="css-1g31t4l-AddLocation"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 448 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
                />
              </svg>
            </button>
          </div>
          <div
            class="css-w40flo-Spinner"
          >
            <div
              class="css-lehx5n-Spinner"
            />
            <div
              class="css-wk0iwg-Spinner"
            />
            <div
              class="css-1xsj7ka-Spinner"
            />
          </div>
          <div
            class="css-gn3t32-Footer"
          >
            <span
              class="css-8ep07x-Footer"
            >
              Results:
            </span>
            <span
              class="css-xp9py7-Footer"
            >
              0 locations
            </span>
          </div>
        </div>
        <div
          aria-labelledby="tab:1:1"
          class="css-818u9c-RightPanel"
          data-reach-tab-panel=""
          hidden=""
          id="panel:1:1"
          role="tabpanel"
          tabindex="-1"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
